src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -991,6 +991,29 @@
         (autosign-certificate-request! subject csr settings)
         (fs/delete (path-to-cert-request csrdir subject))))))
 
+(schema/defn ^:always-validate process-csr-deletion!
+  :- {:outcome (schema/enum :success :not-found :error) :message schema/Str}
```

Comment:
I put a schema on the return value since the consumer of this method is fairly tightly coupled to it's return values and this seemed like a great way to enforce and document the expectations between them. I did not separate out the schema definition into a separate `def` because the data that it returns isn't passed through to other methods or used by other consumers, so it seemed like a pre-mature optimization.

That's _why_ I did what I did, I'm making this note because I'm sure if that's what I should be doing and from the lack of seeing similar instances in the code I'm afraid I'm going against the grain?